### PR TITLE
rubocop config

### PR DESCRIPTION
**Pull request by durandom**
* changes rubocop config to point to manageiq repo instead of manageiq/guides
* don't maintain our own copy of rubocop_cc.yml for codeclimate, but use their support for [external configurations](https://codeclimate.com/changelog/582495c32c33066f1b00191d)

### CODE REVIEW

**Comment by durandom** [comment]
cc @jrafanie

File: .rubocop.yml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 inherit_from:
-- https://raw.githubusercontent.com/ManageIQ/guides/master/.rubocop.yml
+- https://raw.githubusercontent.com/ManageIQ/manageiq/master/.rubocop_base.yml
```
Comment (durandom) [inline .rubocop.yml]: Am I right that we use the `.rubocop_base.yml` from ManageIQ/manageiq rather the one from the guides?

Comment (jrafanie) [inline .rubocop.yml]: Yes @durandom but I think we should be using the local file: `.rubocop_base.yml`, not the URL

Comment (durandom) [inline .rubocop.yml]: But then I have to maintain a style across all repos?
I thought having one in a central place was the point

File: .codeclimate.yml [hunk 2]
```diff
@@ -34,6 +34,12 @@ engines:
   rubocop:
     enabled: true
     config: '.rubocop_cc.yml'
+prepare:
+  fetch:
+  - url: "https://raw.githubusercontent.com/ManageIQ/manageiq/master/.rubocop_base.yml"
+    path: ".rubocop_base.yml"
+  - url: "https://raw.githubusercontent.com/ManageIQ/manageiq/master/.rubocop_cc.yml"
+    path: ".rubocop_cc.yml"
```
Comment (jrafanie) [inline .codeclimate.yml]: Ok, I'm following, so we fetch the _base and _cc from the main repo as part of the codeclimate run... we tell codeclimate to use .rubocop_cc.yml which pulls in the now local .rubocop_base.yml fetched from the main repo.

What do we do about provider changes to rubocop.yml?

File: .rubocop.yml (hunk 1 above)
Comment (jrafanie) [inline .rubocop.yml]: I was misunderstanding... commented above...

File: .codeclimate.yml (hunk 2 above)
Comment (jrafanie) [inline .codeclimate.yml]: in other words, this PR drops all changes on top of the main repos versions... how do we apply any customizations we want for a specific repo?

**Comment by durandom** [comment]
@jrafanie do you think this good, how I split the files? Or do you prefer another way?

**Comment by jrafanie** [comment]
@durandom I think this is good for now, let's see if it works as we want and refine it as we go.

**Comment by jrafanie** [comment]
@durandom Merging, let's see how it goes

**Pull request merged**